internal_project/services/task.py

```diff
@@ -0,0 +1,213 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Akretion (http://www.akretion.com)
+# Benoit Guillot <benoit.guillot@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.addons.component.core import Component
+from odoo.addons.base_rest.components.service import to_bool
+from odoo.exceptions import UserError, MissingError
+import json
+
+
+class ExternalTaskService(Component):
+    _inherit = 'base.rest.service'
+    _name = 'external.task.service'
+    _collection = 'project.project'
+    _usage = 'project_task'
+
+    @property
+    def project(self):
+        return self.work.project
+
+    def search(self, args, offset, limit, order, count):
+        domain = json.loads(args)
+        domain += [('project_id', '=', self.project.id)]
```

Comment:
security issue. If you have from the args the following domain ['|', ['1', '=', '1']] then the domain on project will be by-passed